nabs/preprocessors.py

```diff
@@ -301,4 +302,70 @@ def daq_during_plan():
     return (yield from daq_during_plan())
 
 
+def step_size_decorator(plan):
+    """
+    Grab the last argument (number of steps), and intepret as
+    - step size if float
+    - number of steps if integer
+
+    Only works on step scans in one dimension.
+
+    Parameters
+    ----------
+    plan : plan
+        A bluesky plan that yields bluesky Msg objects.  Must be
+        a scan in one dimension, with the last argument being
+        the number of scan points / step size.
+
+    Returns
+    -------
+    step_size_plan : plan
+        The same plan as before, but modified appropriately to
+        differentiate between step size and number of steps.
+        This will be a callable generator function.
+    """
+
+    @wraps(plan)
+    def inner(*args, **kwargs):
+        if 'num' in kwargs:
+            # Currently unneeded, since daq_ascan and daq_dscan
+            # do not support num kwarg
+            n = kwargs.pop('num')
+        else:
+            # assumes (det_list, motor, start, stop, num)
+            n = args[-1]
```

Comment:
Rather than assume, should we do:

`det_list, motor, start, stop, n = args`

and fail if it doesn't match up with what's expected?